Mainnet/Identity/IdentityContracts/IdentityProvider.cs

```diff
@@ -0,0 +1,143 @@
+﻿using Stratis.SmartContracts;
+
+/// <summary>
+/// A constrained version of ERC780.
+/// This contract only holds claims from one identity provider, the "Owner".
+/// </summary>
+public class IdentityProvider : SmartContract
+{
+    private Address Owner
+    {
+        get
+        {
+            return this.PersistentState.GetAddress(nameof(Owner));
+        }
+        set
+        {
+            this.PersistentState.SetAddress(nameof(Owner), value);
+        }
+    }
+
+    public IdentityProvider(ISmartContractState state) : base(state)
+    {
+        this.Owner = Message.Sender;
+    }
+
+    public void ChangeOwner(Address newOwner)
+    {
+        Assert(this.Owner == Message.Sender);
+        this.Owner = newOwner;
+    }
+
+    public void AddClaim(Address issuedTo, uint topic, byte[] data)
+    {
+        Assert(this.Owner == Message.Sender);
```

Comment:
Enforcing a maximum length on `data` here would be a good idea.

Reply:
What would be max length ?